marlowe/src/Language/Marlowe/Scripts.hs

```diff
@@ -253,10 +253,15 @@ mkMarloweValidator
                         totalIncome = foldMap collectDeposits inputContents
                         totalPayouts = foldMap snd payoutsByParty
                         finalBalance = inputBalance + totalIncome - totalPayouts
-                        -- The total account balance must be paid to a single output to the script.
-                        -- [Marlowe-Cardano Specification: "Constraint 3. Single Marlowe output".]
-                        -- [Marlowe-Cardano Specification: "Constraint 6. Output value to script."]
-                        in checkOwnOutputConstraint marloweData finalBalance
+                        outputBalance = totalBalance (accounts txOutState)
+                        in
+                          -- The total account balance must be paid to a single output to the script.
+                          -- [Marlowe-Cardano Specification: "Constraint 3. Single Marlowe output".]
+                          -- [Marlowe-Cardano Specification: "Constraint 6. Output value to script."]
+                          checkOwnOutputConstraint marloweData finalBalance
+                          -- The value in the script's output UTxO must match the value in the output state.
+                          -- [Marlowe-Cardano Specification: "Constraint 18. Final balance."]
+                          && traceIfFalse "f" (outputBalance == finalBalance)
```

Comment:
The `traceIfFalse` increases the validator size by 90 bytes. The `outputBalance` computation and tests increases the validator size by 132 bytes.